server/routes/updateNewProfile.ts

```diff
@@ -0,0 +1,66 @@
+import { Request, Response, NextFunction } from 'express';
+import { DB } from '../database';
+
+export const Errors = {
+  InvalidUpdate: 'Invalid update',
+  NoProfileFound: 'No profile found',
+  NoAddressFound: 'No address found',
+  NoAddressProvided: 'No address provided in query',
+}
+
+const updateNewProfile = async (
+  models: DB, req: Request, res: Response, next: NextFunction
+) => {
+
+  if (!req.body.address) {
+    return next(new Error(Errors.NoAddressProvided));
+  }
+
+  if (!req.body.email && !req.body.slug && !req.body.name && 
+    !req.body.bio && !req.body.website && !req.body.avatarUrl) {
+    return next(new Error(Errors.InvalidUpdate));
+  }
+
+  const { address, email, slug, name, bio, website, avatarUrl } = req.body;
+
+  const addressModel = await models.Address.findOne({
```

Comment:
To get the address model, we should use `lookupAddressIsOwnedByUser` or similar to ensure that the jwt-identified user actually owns the address in question (or perhaps create a new function that gets the related profile), or else I can modify any profile I want by constructing a query properly.

Reply:
Currently `author_chain` is not present in the request since the profile page operates off of solely the address. I can ensure that the address owner matches the user making the request within the updateNewProfile endpoint. Just let me know if that works!

Reply:
Yep -- as long as we verify that the caller is also the address owner, we're good.